pkgs/os-specific/linux/wireguard/default.nix

```diff
@@ -1,11 +1,17 @@
-{ stdenv, kernel, wireguard-tools, perl }:
+{ stdenv, fetchzip, kernel, perl, wireguard-tools }:
 
 # module requires Linux >= 3.10 https://www.wireguard.io/install/#kernel-requirements
 assert stdenv.lib.versionAtLeast kernel.version "3.10";
 
-stdenv.mkDerivation {
+# wireguard upstreamed since 5.6 https://lists.zx2c4.com/pipermail/wireguard/2019-December/004704.html
+mkIf (stdenv.lib.versionOlder kernel.version "5.6") stdenv.mkDerivation rec {
```

Comment:
`mkIf` only applies for the module system, not our package set.

Reply:
apologies, this is a wip commit, did not mean to push.